swagger_server/itm/itm_scenario_session.py

```diff
@@ -508,9 +508,50 @@ def lookup_probe_response(self, action: Action) -> ProbeResponse:
 
     def update_state(self, action: Action):
         # TODO ITM-69: Update scenario state based on action
-
+        # Look up casualty action is applied to
+        casualty = next((casualty for casualty in self.scenario.state.casualties if casualty.id == action.casualty_id), None)
+        # Check we have a reference to the casualty
+        if casualty:
+            if action.action_type == "APPLY_TREATMENT":
+                # using getattr in case treatment not provided as parameter
+                supplies_used = getattr(action.parameters, 'treatment', None)
+                if supplies_used in self.scenario.state.supplies:
+                    # removing one instance of the supplies_used e.g Tourniquet from supplies list
+                    self.scenario.state.supplies.remove(supplies_used)
+                else:
+                    print(f"{supplies_used} is not found in the supplies list. (Possible that no parameter was supplied)")
+                
+                # remove injury from casualty
+                for injury in casualty.injuries:
+                    if injury.location == action.parameters.location:
+                        casualty.remove(injury)
+                        break
+            
+            # if tagging a casualty then update the tag to the category parameter
+            if action.action_type == "TAG_CASUALTY":
+                 # getattr to account for partially specified action. If they don't tell us what to change the tag to keep it the same
+                tag = getattr(action.parameters, 'category', casualty.tag)
+                self.tag_casualty(self.session_id, casualty.id, tag)
+            
```

Comment:
I'm not seeing the following from the ticket:
- Flip Casualty flags if necessary based on action (e.g., assessed , ~~tag~~)
- Flip Vitals flags if necessary based on action (e.g., conscious, responsive, breathing)

Reply:
I thought by calling tag_casualty the flag would be set. In that function casualty.tag gets set

Reply:
`for casualty in self.scenario.state.casualties:
            if casualty.id == casualty_id:
                casualty.tag = tag
                response = self.scenario.state
                self._add_history(
                    "Tag Casualty",
                    {"Session ID": self.session_id, "Casualty ID": casualty_id, "Tag": tag},
                    response.to_dict())
                return response`

Reply:
Yes, that's why I crossed out the word "tag" in my comment.  Everything else in those two bullets needed to be addressed, I believe  (assessed, conscious, responsive, breating).